test/e2e/multikueue/e2e_test.go

```diff
@@ -270,6 +273,84 @@ var _ = ginkgo.Describe("MultiKueue", func() {
 			})
 		})
 
+		ginkgo.It("Should create pods on worker if the deployment is admitted", func() {
+			deployment := testingdeployment.MakeDeployment("deployment", managerNs.Name).
+				Image(util.E2eTestSleepImage, []string{"10m"}).
+				Replicas(3).
+				Queue(managerLq.Name).
+				Obj()
+
+			ginkgo.By("Creating the deployment", func() {
+				gomega.Expect(k8sManagerClient.Create(ctx, deployment)).Should(gomega.Succeed())
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdDeployment := &appsv1.Deployment{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).To(gomega.Succeed())
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Wait for replicas ready", func() {
+				createdDeployment := &appsv1.Deployment{}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), createdDeployment)).To(gomega.Succeed())
+					g.Expect(createdDeployment.Status.ReadyReplicas).To(gomega.Equal(int32(3)))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			pods := &corev1.PodList{}
+			gomega.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Namespace),
+				client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
+
+			// Check all worker pods are in a running phase.
+			ensurePodWorkloadsRunning(pods, *managerNs, multiKueueAc)
+
+			deploymentConditions := []appsv1.DeploymentCondition{
+				{
+					Type:    appsv1.DeploymentAvailable,
+					Status:  corev1.ConditionTrue,
+					Reason:  "MinimumReplicasAvailable",
+					Message: "Deployment has minimum availability.",
+				},
+				{
+					Type:   appsv1.DeploymentProgressing,
+					Status: corev1.ConditionTrue,
+					Reason: "NewReplicaSetAvailable",
+				},
+			}
+
+			ginkgo.By("Waiting for the deployment to get status updates", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					createdDeployment := appsv1.Deployment{}
+					g.Expect(k8sManagerClient.Get(ctx, client.ObjectKeyFromObject(deployment), &createdDeployment)).To(gomega.Succeed())
+					g.Expect(createdDeployment.Status.ReadyReplicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.Replicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.UpdatedReplicas).To(gomega.Equal(int32(3)))
+					g.Expect(createdDeployment.Status.Conditions).To(gomega.BeComparableTo(deploymentConditions, cmpopts.IgnoreFields(appsv1.DeploymentCondition{}, "LastTransitionTime", "LastUpdateTime", "Message"))) // Ignoring message as it is required to gather the pod replica set
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Increasing the replica count on the deployment", func() {
```

Comment:
Can we also a test for scaling down by ensuring that there is no race condition when replica count is reduced.